dev/TreeView/APITests/TreeViewTests.cs

```diff
@@ -501,6 +502,171 @@ public void VerifyVisualTree()
+
+                treeView.ItemsSource = null;
+
+                Verify.IsNotNull(selectionChangedEventArgs);
+                Verify.AreEqual(0, selectionChangedEventArgs.AddedItems.Count);
+                Verify.AreEqual(1, selectionChangedEventArgs.RemovedItems.Count);
+                Verify.AreEqual(1, selectionChangedEventArgs.RemovedItems[0]);
+            });
+        }
+
+        [TestMethod]
+        public void MultipleSelectionChanged()
+        {
+            RunOnUIThread.Execute(() =>
+            {
+                // input data:
+                // - 1
+                //   - 11
+                //   - 12
+                //   - 13
+                // - 2
+                //   - 21
+                // - 3
+                TreeViewSelectionChangedEventArgs selectionChangedEventArgs = null;
+
+                var treeView = new TreeView { SelectionMode = TreeViewSelectionMode.Multiple };
+                treeView.SelectionChanged += (s, e) => selectionChangedEventArgs = e;
+
+                var node1 = new TreeViewNode { Content = "1", IsExpanded = true };
+                var node11 = new TreeViewNode { Content = "11" };
+                var node12 = new TreeViewNode { Content = "12" };
+                var node13 = new TreeViewNode { Content = "13" };
+                node1.Children.Add(node11);
+                node1.Children.Add(node12);
+                node1.Children.Add(node13);
+
+                var node2 = new TreeViewNode { Content = "2", IsExpanded = true };
+                var node21 = new TreeViewNode { Content = "21" };
+                node2.Children.Add(node21);
+
+                var node3 = new TreeViewNode { Content = "3" };
+
+                treeView.RootNodes.Add(node1);
+                treeView.RootNodes.Add(node2);
+                treeView.RootNodes.Add(node3);
+                Content = treeView;
+                Content.UpdateLayout();
+
+                var tvi1 = (TreeViewItem)treeView.ContainerFromItem(node1);
+                var tvi11 = (TreeViewItem)treeView.ContainerFromItem(node11);
+                var tvi12 = (TreeViewItem)treeView.ContainerFromItem(node12);
+                var tvi13 = (TreeViewItem)treeView.ContainerFromItem(node13);
+                var tvi2 = (TreeViewItem)treeView.ContainerFromItem(node2);
+                var tvi21 = (TreeViewItem)treeView.ContainerFromItem(node21);
+                var tvi3 = (TreeViewItem)treeView.ContainerFromItem(node3);
+
+                // - 1         selected
+                //   - 11      selected
+                //   - 12      selected
+                //   - 13      selected
+                // - 2
+                //   - 21
+                // - 3
+                tvi1.IsSelected = true;
+
+                Verify.IsNotNull(selectionChangedEventArgs);
+                Verify.AreEqual(4, selectionChangedEventArgs.AddedItems.Count);
+                Verify.AreEqual(node1, selectionChangedEventArgs.AddedItems[0]);
```

Comment:
I am not sure if we should rely on the items being in that exact order in the event args or if we should check if it is just in the list.

Reply:
Good point. You're right I think the order is an implementation details of the treeview. I will switch to `Contains` instead.